roles/set_custom_fact/tasks/main.yml

```diff
@@ -145,3 +129,14 @@
 - name: Set variable generic_project_name
   set_fact: generic_project_name={{ ansible_hostname | regex_replace('-\d+', '') }}
   tags: [ 'master', 'custom', 'rename', 'update','idc_import','package_management','kalite_import' ]
+
+- name: Stat /dev/sda1
+  stat: path=/dev/sda1
+  register: sda1
+  tags: ['always']
+
+- name: Use sda1 as external hard drive if sda1 exists and on arm architecture
+  set_fact: ext_hdd={{ (ansible_architecture == "armhf" or ansible_architecture == "armv7l") and (sda1.stat.isblk | default (omit)) == True }}
+  when: firststart.stat.exists == false or firststart is undefined
```

Comment:
doesn't set_fact store the variable in custom fact ? how can I do it otherwise ?